cavern/src/intTest/java/org/opencadc/cavern/FilesPutTest.java

```diff
@@ -0,0 +1,318 @@
+        putTestURIStr = baseURI + "/filesPutTest-" + System.currentTimeMillis();
+
+        log.debug("test dir base URI: " + baseURI);
+        log.debug("put test folder: " +  putTestFolderURI);
+
+        putTestFolderURI = new VOSURI(new URI(putTestURIStr));
+        final URI serviceURI = putTestFolderURI.getServiceURI();
+
+        log.debug("nodes serviceURI: " + putTestFolderURI);
+
+        RegistryClient rc = new RegistryClient();
+        nodesURL = rc.getServiceURL(serviceURI, Standards.VOSPACE_NODES_20, AuthMethod.CERT);
+
+        // Set up the put test container folder
+        Subject.doAs(cadcauthSubject, new PrivilegedExceptionAction() {
+            @Override
+            public Object run() throws Exception {
+                ContainerNode cn = new ContainerNode(putTestFolderURI);
+                putNode(new URL(nodesURL.toExternalForm() + cn.getUri().getPath()), cn);
+                return null;
+            }
+        });
+    }
+
+    @Test
+    public void testPutFileOK() throws Throwable {
+        final String uri = putTestURIStr + "/smallTextFile.rtf";
+        final VOSURI nodeURI = new VOSURI(uri);
+
+        try {
+            Subject.doAs(cadcauthSubject, new PrivilegedAction<Object>() {
+                public Object run() {
+                    try {
+                        String filename = "src/test/resources/smallTextFile.rtf";
+                        putFile(nodeURI, filename);
+                        return null;
+                    } catch (Throwable t) {
+                        throw new RuntimeException(t);
+                    }
+                }
+            });
+            
+        } catch (Throwable t) {
+            log.error(t.getMessage(), t);
+            Assert.fail("Unexcepted exception: " + t.getMessage());
+        }
+    }
+
+
+    @Test
+    public void testPutOverwriteFileOK() throws Throwable {
+        final String uri = putTestURIStr + "/smallTextFile2.rtf";
+        final VOSURI nodeURI = new VOSURI(uri);
+
+        // Make a DataNode of the same name as the test file,
+        // PUT that node using /cavern/nodes service
+        Subject.doAs(cadcauthSubject, new PrivilegedExceptionAction() {
+            @Override
+            public Object run() throws Exception {
+                DataNode dn = new DataNode(nodeURI);
+                dn.getProperties().add(new NodeProperty(VOS.PROPERTY_URI_ISPUBLIC, Boolean.FALSE.toString()));
+                putNode(new URL(nodesURL.toExternalForm() + dn.getUri().getPath()), dn);
+                return null;
+            }
+        });
+
+        // Now attempt to PUT the file using /cavern/files - should still work
+        try {
+            Subject.doAs(cadcauthSubject, new PrivilegedAction<Object>() {
+                public Object run() {
+                    try {
+                        String filename = "src/test/resources/smallTextFile2.rtf";
+                        putFile(nodeURI, filename);
```

Comment:
After the PUT, a GET should be performed and the filename, content-type, file-size, md5sum should all be checked for correctness.

Reply:
done